### DISCUSSION TITLE
Wish to have OSC9 disabled on remote shells only -- how best to do it?

### DISCUSSION THREAD

**Discussion by darkvertex** [JanDeDobbeleer/oh-my-posh#3224] [Troubleshoot]
So I have this situation where Windows Terminal supports OSC9 and remembers where I am, which is cool when I'm somewhere local like in WSL Linux or in Powershell on my local Windows, but when I'm ssh'ing to a remote computer (where I also configured oh-my-posh), OSC9 codes get sent also.

When I restart Windows Terminal and it remembers the tabs, it tries to cd to the last seen folder, which is of course non-existent for a remote ssh session.

So I'm wondering what's the most elegant way to enable OSC9 support only for local shells and not remote ones? (If not possible, I'll be glad to contribute some kind of detection feature.)

**Comment by JanDeDobbeleer** [comment]
The `pwd` config value allows templating. This means you can identify whether or not you're in an SSH session by leveraging [environment](https://ohmyposh.dev/docs/configuration/templates#environment-variables) or other logic. [These](https://github.com/JanDeDobbeleer/oh-my-posh/blob/dda12ff877de1946ad1def1c5955ba3c1f67c871/src/segments/session.go#L33) are the variables we use in the session segment.
